### PR TITLE
Issue #693: Emit phase_start/phase_complete in run-*.sh for single-issue /auto

### LINKED ISSUE #693: run-*.sh: 単一 Issue /auto でも phase_start/phase_complete event を emit して rollup を機能させる
## 背景

`scripts/auto-events-rollup.sh` (#638) は `.tmp/auto-events.jsonl` から daily rollup を生成するが、**セッション表 / Phase Distribution / Recovery Tier 集計はすべて `phase_start` / `phase_complete` event に依存**している。

しかし `phase_start` / `phase_complete` を emit するのは `scripts/run-auto-sub.sh` (--batch mode 専用) のみで、単一 Issue `/auto` の `run-code.sh` / `run-review.sh` / `run-merge.sh` 個別実行では emit されない。

### 観察された事例

- 2026-06-17 セッション (35207-1781711844 + 73050-1781700419 + 68137-1781707036 等): 単一 Issue /auto を複数完走したが、`phase_start` / `phase_complete` event は 0 件
- 6/17 の events 23 件は ci_wait (9) / max_silent_window (13) / test_result (1) のみ
- 2026-06-18 にバックフィル生成した `docs/reports/auto-events-rollup-2026-06-17.md` は **全セクション空** (commit 62c24d7)

### 影響

- 単一 Issue /auto (実際の使われ方の主流) では rollup が常に空 → セッション集計の網羅性が `--batch` のみに限定
- Phase Distribution の中央値/p95 などの統計が batch run のみで歪む
- `#658` (rollup 自動実行) を整備しても data が空のため運用価値が低い
- 同様に `token_usage` event も `run-auto-sub.sh` のみ emit のため、#662 観測も batch でしか機能しない

### 現状の部分実装

`#672` で `_maybe_emit_phase_complete` backfill 関数が各スクリプトに追加済みだが、以下の不足により機能していない:

- `EMIT_ISSUE_NUMBER` / `EMIT_PHASE_NAME` が各スクリプト内で未設定 → emit 呼び出しが `issue: 0` になる
- `emit_event "phase_start"` 呼び出しが存在しない → backfill の条件 `_last_event == "phase_start"` が満たされない
- `run-merge.sh` では `test_result` emit 後に EXIT するため、backfill も機能しない

## 目的

単一 Issue `/auto` でも rollup が意味のあるセッション集計を出せるよう、phase boundary event の emit 経路を拡張する。

## 提案

候補 A (推奨): `run-code.sh` / `run-review.sh` / `run-merge.sh` 各冒頭で `phase_start`、末尾で `phase_complete` を emit
- 既存の `wait-ci-checks.sh` が `ci_wait` を emit するのと同じパターン
- `emit-event.sh` を source して `emit_event "phase_start" "phase=<name>"` 追加
- 副作用なし、wrapper exit code に影響なし

候補 B: rollup script 側で `ci_wait` / `test_result` から phase boundary を推定
- run-*.sh 変更不要だが、推定ロジックが複雑化 (例: ci_wait は review 直前、test_result は merge 直後 etc.)
- 既存 batch 経由の正確な data と混在すると semantic がずれる

候補 A が **シンプル & 一貫性高い**。`token_usage` も同様に各 run-*.sh で emit する拡張に発展可能 (別 Issue 検討)。

### 実装上の注意点 (auto-resolved)

**二重 emit ガード**: `run-auto-sub.sh` 経由で呼び出された場合、`EMIT_PHASE_NAME` は既設定かつ `phase_start` は既 emit 済み。`EMIT_PHASE_NAME` が未設定の場合のみ (= 単一 Issue 直接呼び出し) 設定 + emit を行うことで二重 emit を回避する。

**phase 名**: `run-auto-sub.sh` の呼び出し規約に合わせ:
- `run-code.sh`: `--pr` フラグ → `"code-pr"`, `--patch` フラグ → `"code-patch"`, フラグなし → `"code"`
- `run-review.sh`: `"review"`
- `run-merge.sh`: `"merge"`

**phase_complete の明示 emit**: `run-merge.sh` は `emit_event "test_result"` 後に EXIT するため backfill 条件 `_last_event == "phase_start"` が成立しない。`run-auto-sub.sh` と同様に正常終了時は明示的に `emit_event "phase_complete"` を呼ぶ。

### Pre-merge (auto-verified)

- [x]  `scripts/run-code.sh` に `emit_event "phase_start"` 呼び出しが追加されている
- [x]  `scripts/run-code.sh` に `emit_event "phase_complete"` 呼び出しが追加されている
- [x]  `scripts/run-code.sh` に `EMIT_ISSUE_NUMBER` の設定が追加されている
- [x]  `scripts/run-code.sh` に `EMIT_PHASE_NAME` の設定が追加されている
- [x]  `scripts/run-review.sh` に `emit_event "phase_start"` 呼び出しが追加されている
- [x]  `scripts/run-review.sh` に `emit_event "phase_complete"` 呼び出しが追加されている
- [x]  `scripts/run-review.sh` に `EMIT_ISSUE_NUMBER` の設定が追加されている
- [x]  `scripts/run-review.sh` に `EMIT_PHASE_NAME` の設定が追加されている
- [x]  `scripts/run-merge.sh` に `emit_event "phase_start"` 呼び出しが追加されている
- [x]  `scripts/run-merge.sh` に `emit_event "phase_complete"` 呼び出しが追加されている
- [x]  `scripts/run-merge.sh` に `EMIT_ISSUE_NUMBER` の設定が追加されている
- [x]  `scripts/run-merge.sh` に `EMIT_PHASE_NAME` の設定が追加されている
- [x]  rubric 基準を満たす
- [x]  backfill 関数が EMIT_PHASE_NAME ガードを保持している
- [x]  既存 bats テストが green

### Pre-merge: Sessions fallback (auto-verified)

- [x]  `scripts/auto-events-rollup.sh` の Sessions ロジックが `phase_start` を参照している
- [x]  `scripts/auto-events-rollup.sh` の Sessions ロジックが `phase_complete` を参照している
- [x]  `tests/auto-events-rollup.bats` に fallback 用の新規テストが追加されている
- [x]  `tests/auto-events-rollup.bats` に incomplete outcome 用の新規テストが追加されている
- [x]  `bats tests/auto-events-rollup.bats` がすべて green

### Post-merge

- [x] 次回 単一 Issue pr-route `/auto` 完走後の rollup で Sessions テーブルが空でなく phase_start/phase_complete 経由のセッションが集計されることを確認 
- [x] 次回 単一 Issue `/auto` 完走後の rollup で Sessions テーブルに `| #N` 行が出力されることを確認 

## 自動解決済み曖昧ポイント

| # | 曖昧ポイント | 解決方針 | 根拠 |
|---|-------------|---------|------|
| 1 | `phase_complete` は backfill に任せるか明示 emit するか | 明示 emit を追加 (backfill は crashフォールバック扱い) | `run-merge.sh` では `test_result` emit 後に EXIT するため backfill 条件が不成立。`run-auto-sub.sh` も明示 emit を使用 |
| 2 | `run-auto-sub.sh` 経由呼び出し時の二重 emit | `EMIT_PHASE_NAME` が未設定の場合のみ emit (ガード追加) | `run-auto-sub.sh` は `export EMIT_PHASE_NAME="$phase"` 後に runner_script を呼び出す。この値が引き継がれる |
| 3 | phase 名の命名規則 | `run-auto-sub.sh` の命名に合わせる (code-pr/code-patch/review/merge) | `run-auto-sub.sh:239,247,257,260` を参照 |

## 関連

- 発生元: 2026-06-18 rollup 欠損調査 (parent session 手動バックフィル commit 62c24d7)
- 関連: #658 (rollup 自動実行 — 本 Issue で data 整備)
- 関連: #638 (auto-events-rollup.sh 実装)
- 関連: #662 (token_usage observation — 同じく run-auto-sub.sh のみ emit、本 Issue と類似構造)
- 関連: #672 (phase_complete backfill 追加 — 本 Issue の部分実装前提)
- 参照: `scripts/auto-events-rollup.sh:140,153,158` (phase_complete/phase_start 依存箇所)
- 参照: `scripts/run-auto-sub.sh` (現状唯一の phase_start/phase_complete emitter)

### PR DESCRIPTION
**Pull request by saito**
## Summary

`run-code.sh` / `run-review.sh` / `run-merge.sh` に `phase_start` / `phase_complete` イベントの emit を追加。

- `auto-events-rollup.sh` のセッション集計はこれらのイベントに依存するが、単一 Issue `/auto` では `run-auto-sub.sh` を経由しないため emit されていなかった
- EMIT ガードブロック (`EMIT_PHASE_NAME` 未設定の場合のみ emit) で `run-auto-sub.sh` 経由時の二重 emit を回避
- `run-merge.sh` は `test_result` emit 後に EXIT するため backfill 条件が不成立になる問題を明示 emit で解決
- 各スクリプトに bats テスト 3 件追加 (phase_start emit / 二重 emit ガード / phase_complete emit)

## Verification (pre-merge)

- `grep "emit_event.*phase_start" scripts/run-code.sh` PASS
- `grep "emit_event.*phase_complete" scripts/run-code.sh` PASS
- `grep "EMIT_ISSUE_NUMBER=" scripts/run-code.sh` PASS
- `grep "EMIT_PHASE_NAME=" scripts/run-code.sh` PASS
- 同様に `run-review.sh` / `run-merge.sh` も全 PASS
- rubric: 3 スクリプト全で EMIT ガード / phase_complete 明示 emit / phase 名一致 PASS
- `section_contains` で backfill 関数の EMIT_PHASE_NAME ガード保持確認 PASS
- `bats tests/run-code.bats tests/run-review.bats tests/run-merge.bats` → 74 tests PASS

## Verification (post-merge)

- 次回 単一 Issue pr-route `/auto` 完走後の rollup で Sessions テーブルが空でないことを確認

closes #693

### CODE REVIEW

File: scripts/run-code.sh [hunk 1]
```diff
@@ -234,6 +248,10 @@ if [[ "$ROUTE_FLAG" == "--pr" && $EXIT_CODE -eq 0 ]]; then
   fi
 fi
 
+if [[ $EXIT_CODE -eq 0 ]]; then
+  emit_event "phase_complete" "phase=${EMIT_PHASE_NAME}"
+fi
```
Comment (saito) [inline scripts/run-code.sh:255]: **[Edge Case/Robustness] Double phase_complete when called via run-auto-sub.sh**

`phase_start` の emit は `EMIT_PHASE_NAME` 未設定ガードで保護されているが、このブロックは同じガードを持たない。`run-auto-sub.sh` 経由 (--batch / XL sub-issue) で呼び出された場合、`run-auto-sub.sh` の `run_phase_with_recovery` も同 phase/issue の `phase_complete` を emit するため、`auto-events.jsonl` に二重エントリが生成される。

影響: `auto-events-rollup.sh` がフェーズシーケンス列を二重計上し、Phase Distribution の duration 行が重複する。

推奨修正: guard block 内でローカルセンチネル変数を設定し、このブロックをセンチネルで条件分岐する:
```bash
_EMIT_PHASE_OWNED=""
if [[ -z "${EMIT_PHASE_NAME:-}" ]]; then
  _EMIT_PHASE_OWNED=1
  ...
fi
# ...
if [[ $EXIT_CODE -eq 0 && -n "${_EMIT_PHASE_OWNED:-}" ]]; then
  emit_event "phase_complete" "phase=${EMIT_PHASE_NAME}"
fi
```

Severity: SHOULD — batch/XL path を破壊するが、単一 Issue /auto (本 Issue の修正対象) は影響なし

File: scripts/run-review.sh [hunk 2]
```diff
@@ -145,6 +153,10 @@ if [[ $EXIT_CODE -eq 143 || $EXIT_CODE -eq 0 ]]; then
   fi
 fi
 
+if [[ $EXIT_CODE -eq 0 ]]; then
+  emit_event "phase_complete" "phase=${EMIT_PHASE_NAME}"
```
Comment (saito) [inline scripts/run-review.sh:159]: **[Edge Case/Robustness] Double phase_complete — run-review.sh も同様**

`run-code.sh` と同じ問題。`phase_complete` emit が `EMIT_PHASE_NAME` ガードなしのため、`run-auto-sub.sh` 経由時に二重 emit が発生する。

Severity: SHOULD

File: scripts/run-merge.sh [hunk 3]
```diff
@@ -169,6 +177,10 @@ if [[ $EXIT_CODE -eq 0 && -n "${AUTO_EVENTS_LOG:-}" ]]; then
   fi
 fi
 
+if [[ $EXIT_CODE -eq 0 ]]; then
+  emit_event "phase_complete" "phase=${EMIT_PHASE_NAME}"
```
Comment (saito) [inline scripts/run-merge.sh:183]: **[Edge Case/Robustness] Double phase_complete — run-merge.sh も同様**

`run-code.sh` と同じ問題。`phase_complete` emit が `EMIT_PHASE_NAME` ガードなしのため、`run-auto-sub.sh` 経由時に二重 emit が発生する。`run-merge.sh` は `test_result` emit 後に `phase_complete` を出す構造のため、rollup 側でも二重計上が確実に発生する。

Severity: SHOULD

File: tests/run-code.bats [hunk 4]
```diff
@@ -427,3 +427,36 @@ MOCK
...
+}
+
+@test "emit: phase_start not emitted when EMIT_PHASE_NAME is pre-set (no double emit)" {
+    EMIT_LOG="$BATS_TEST_TMPDIR/emit.log"
+    cat > "$MOCK_DIR/emit-event.sh" <<MOCK
+emit_event() { echo "\$@" >> "${EMIT_LOG}"; }
+MOCK
+    export EMIT_PHASE_NAME="code-pr"
+    run bash "$SCRIPT" 123 --pr
+    unset EMIT_PHASE_NAME
```
Comment (saito) [inline tests/run-code.bats:449]: **[Test Coverage Gap] "no double emit" テストが phase_complete を検証していない**

このテストは `phase_start` が emit されないことは確認するが、`phase_complete` も同様にガードされているか検証しない。以下のアサーションの追加を推奨:

```bash
! grep -q '"event":"phase_complete"' "$EMIT_LOG"
```

またはシンプルに:
```bash
! grep -q 'phase_complete' "$EMIT_LOG"
```

同様に `tests/run-review.bats`・`tests/run-merge.bats` の対応テストにも追加が必要。

Severity: SHOULD

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| 条件 | 結果 | 備考 |
|------|------|------|
| run-code.sh に `emit_event "phase_start"` 追加 | PASS | — |
| run-code.sh に `emit_event "phase_complete"` 追加 | PASS | — |
| run-code.sh に `EMIT_ISSUE_NUMBER` 設定 | PASS | — |
| run-code.sh に `EMIT_PHASE_NAME` 設定 | PASS | — |
| run-review.sh に `emit_event "phase_start"` 追加 | PASS | — |
| run-review.sh に `emit_event "phase_complete"` 追加 | PASS | — |
| run-review.sh に `EMIT_ISSUE_NUMBER` 設定 | PASS | — |
| run-review.sh に `EMIT_PHASE_NAME` 設定 | PASS | — |
| run-merge.sh に `emit_event "phase_start"` 追加 | PASS | — |
| run-merge.sh に `emit_event "phase_complete"` 追加 | PASS | — |
| run-merge.sh に `EMIT_ISSUE_NUMBER` 設定 | PASS | — |
| run-merge.sh に `EMIT_PHASE_NAME` 設定 | PASS | — |
| rubric: 3スクリプト全ガード/phase_complete/phase名一致 | PASS | — |
| backfill 関数が EMIT_PHASE_NAME ガード保持 | PASS | — |
| bats テスト green | PASS | CI "Run bats tests" SUCCESS で代替検証 |
| (post-merge) rollup Sessions 集計確認 | POST-MERGE | 次回 /auto 完走後に観測 |

## CI Status

| ジョブ | ステータス | 備考 |
|--------|-----------|------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### Issue Count Summary

- Line comments: 4 (MUST: 0, SHOULD: 4, CONSIDER: 0)
- General Comments: 0

---
Next: run `/merge 697`.

**Comment by saito** [comment]
Review: phase_complete の二重 emit が run-auto-sub.sh 経由時に発生する。EMIT_PHASE_NAME ガードブロックは phase_start を保護しているが phase_complete は同じガードを持たないため、batch/XL path で auto-events.jsonl に二重エントリが生成される。センチネル変数で修正予定。

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response (review-light — All 4 Perspectives)

| アスペクト | ファイル | 行 | 問題 | 対応 | 修正/スキップ理由 |
|-----------|---------|-----|------|------|----------------|
| Edge Case/Robustness | scripts/run-code.sh | 253 | phase_complete が run-auto-sub.sh 経由時に二重 emit される | 修正済み | `_EMIT_PHASE_OWNED` センチネル変数を導入し phase_start guard と同じ条件で phase_complete を制御 |
| Edge Case/Robustness | scripts/run-review.sh | 157 | 同上 | 修正済み | 同上 |
| Edge Case/Robustness | scripts/run-merge.sh | 181 | 同上 | 修正済み | 同上 |
| Test Coverage Gap | tests/run-code.bats | 449 | "no double emit" テストが phase_complete を検証していない | 修正済み | `! grep -q "phase_complete"` アサーション追加 |
| Test Coverage Gap | tests/run-review.bats | — | 同上 | 修正済み | 同上 |
| Test Coverage Gap | tests/run-merge.bats | — | 同上 | 修正済み | 同上 |

### Lightweight Re-check Results

- validate-skill-syntax.py: PASS (0 errors, 0 warnings)
- Fix diff check: PASS
- New issues: なし

### Acceptance Criteria Updates

変更なし (ポリシー変更なし)
